src/client/executor.rs

```diff
@@ -397,6 +398,30 @@ impl Client {
                     implicit_session,
                 },
                 Err(mut err) => {
+                    // If the error is a reauthentication required error, we reauthenticate and
+                    // retry the operation.
+                    if err.is_reauthentication_required() {
+                        let credential = self.inner.options.credential.as_ref().ok_or(
```

Comment:
both of these errors should be impossible under the current sever spec, but this does future proof us.